### PR TITLE
Deploy addon agent

### PR DESCRIPTION
**Pull request by qiujian16**
Signed-off-by: Jian Qiu <jqiu@redhat.com>

### CODE REVIEW

**Comment by qiujian16** [comment]
https://github.com/open-cluster-management/backlog/issues/15781

File: pkg/addon/manifests/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          secretName: {{ .KubeConfigSecret }}
+      containers:
+      - name: klusterlet-addon-workmgr
+        image: {{ .Image }}
+        args:
+          - "/agent"
+          - "--hub-kubeconfig=/var/run/hub/kubeconfig"
+          - "--cluster-name={{ .ClusterName }}"
+          - "--port=4443"
+          - "--agent-address=klusterlet-addon-workmgr.open-cluster-management-agent.svc"
```
Comment (zhiweiyin318) [inline pkg/addon/manifests/deployment.yaml:33]: there are 3 cases here:
1. --agent-address: use svc in local-cluster
2. --agent-route: route on OCP
3. --agent-servic:  loadbalancer

refer: https://github.com/open-cluster-management/klusterlet-addon-operator/blob/020356206ba0e26153c8bdb54953012f6dab829e/helm-charts/work-manager/templates/deployment.yaml#L55

File: pkg/addon/manifests/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      serviceAccountName: klusterlet-addon-workmgr-sa
+      securityContext:
+        runAsNonRoot: true
+      volumes:
+      - name: hub-config
+        secret:
+          secretName: {{ .KubeConfigSecret }}
+      containers:
+      - name: klusterlet-addon-workmgr
+        image: {{ .Image }}
```
Comment (zhiweiyin318) [inline pkg/addon/manifests/deployment.yaml:27]: should we add an image pull secret here or add it into sa ?

**Pull request merged**